crates/transpiler/src/passes/basis_translator/mod.rs

```diff
@@ -223,87 +215,59 @@ pub fn run_basis_translator(
 }
 
 /// Method that extracts all gate instances identifiers from a DAGCircuit.
-fn extract_basis(
-    py: Python,
-    circuit: &DAGCircuit,
-    min_qubits: usize,
-) -> PyResult<IndexSet<GateIdentifier, ahash::RandomState>> {
-    let mut basis = IndexSet::default();
+fn extract_basis(circuit: &DAGCircuit, min_qubits: usize) -> AhashIndexSet<GateIdentifier> {
+    let mut basis = AhashIndexSet::default();
     // Recurse for DAGCircuit
     fn recurse_dag(
-        py: Python,
         circuit: &DAGCircuit,
-        basis: &mut IndexSet<GateIdentifier, ahash::RandomState>,
+        basis: &mut AhashIndexSet<GateIdentifier>,
         min_qubits: usize,
-    ) -> PyResult<()> {
+    ) {
         for (_node, operation) in circuit.op_nodes(true) {
             if circuit.get_qargs(operation.qubits).len() >= min_qubits {
                 basis.insert((operation.op.name().to_string(), operation.op.num_qubits()));
             }
             if operation.op.control_flow() {
-                let OperationRef::Instruction(inst) = operation.op.view() else {
-                    unreachable!("Control flow operation is not an instance of PyInstruction.")
-                };
-                let inst_bound = inst.instruction.bind(py);
-                for block in inst_bound.getattr("blocks")?.try_iter()? {
-                    recurse_circuit(py, block?, basis, min_qubits)?;
+                for block in operation.op.blocks() {
+                    recurse_circuit(&block, basis, min_qubits);
                 }
             }
         }
-        Ok(())
     }
 
     // Recurse for QuantumCircuit
```

Comment:
```suggestion
    // Recurse for CircuitData
```
.... or just delete, the function name seems clear enough 😉

Reply:
Fixed in 7cb647cdd4e84ce1b767112262edb910208f5ffb